spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerMappingContext.java

```diff
@@ -75,7 +91,9 @@ public FieldNamingStrategy getFieldNamingStrategy() {
 
   protected <T> SpannerPersistentEntityImpl<T> constructPersistentEntity(
       TypeInformation<T> typeInformation) {
-    return new SpannerPersistentEntityImpl<>(typeInformation);
+    return new SpannerPersistentEntityImpl<>(typeInformation, this,
+        this.applicationContext == null ? new ConverterAwareMappingSpannerEntityProcessor(this)
+            : this.applicationContext.getBean(SpannerEntityProcessor.class));
```

Comment:
Expanded this into if-else blocks. 
I thought of getting rid of the if-else block completely and replace with non-null checks, my reasonings:
`this.applicationContext` is set from `ApplicationContextAware` in `SpannerContextMapping` bean post creation process.
So by the time `constructPersistentEntity()` is called, we are in post process of entity bean creations, and `this.applicationContext` and `this.applicationContext.getBean(SpannerEntityProcessor.class)` should both be present already. 
But eventually dial back to the current if-else because the change breaks a lot of unit tests and setting up mock applicationContext only fixes some of them. So I feel that this removal probably should belong to a separate future pr along with some test cleanups. what are your thoughts?

Reply:
Yeah, keep it. It's easy to follow now, and prevents future issues.